truth_set/java/structure/organisation/complex-code/ECommerce.java

```diff
@@ -0,0 +1,224 @@
+    public void setItems(List<OrderItem> i) { this.items = i; }
+    public Shipping getShipping() { return shipping; }
+    public void setShipping(Shipping s) { this.shipping = s; }
+    public List<Coupon> getCoupons() { return coupons; }
+    public void setCoupons(List<Coupon> c) { this.coupons = c; }
+}
+
+public class ECommerce {
+    public Map<String, Object> processOrder(OrderInput order) {
+        Map<String, Object> finalOrder = new HashMap<>();
+        if (order == null || order.getCustomer() == null || order.getCustomer().getId() == null || 
+            order.getItems() == null || order.getItems().isEmpty()) {
+            return null;
+        }
+
+        finalOrder.put("customerId", order.getCustomer().getId());
+        finalOrder.put("customerName", order.getCustomer().getName() != null ? 
+                      order.getCustomer().getName() : "Valued Customer");
+
+        if (order.getCustomer().getLoyaltyStatus() != null && 
+            order.getCustomer().getLoyaltyStatus() != LoyaltyStatus.NONE) {
+            finalOrder.put("loyalty", "Status: " + order.getCustomer().getLoyaltyStatus());
+        }
+
+        Map<String, Object> items = new HashMap<>();
+        finalOrder.put("items", items);
+
+        for (int i = 0; i < order.getItems().size(); i++) {
+            OrderItem item = order.getItems().get(i);
+            if (item.getProductId() == null || item.getQuantity() <= 0) {
+                continue;
+            }
+
+            double itemTotal = 0;
+            Double productPrice = getProductPrice(item.getProductId());
+            if (productPrice == null) {
+                items.put("item_" + i, "Price Not Available");
+                continue;
+            }
+
+            itemTotal += productPrice * item.getQuantity();
+
+            if (item.getOptions() != null) {
+                for (Option option : item.getOptions()) {
+                    if (option.getOptionId() == null || option.getValue() == null) {
+                        continue;
+                    }
+                    Double optionPrice = getOptionPrice(option.getOptionId(), option.getValue());
+                    if (optionPrice != null) {
+                        itemTotal += optionPrice;
+                    }
+                }
+            }
+
+            if (item.getDiscounts() != null) {
+                for (Discount discount : item.getDiscounts()) {
+                    if (discount.getDiscountId() == null || discount.getAmount() <= 0) {
+                        continue;
+                    }
+                    itemTotal -= discount.getAmount();
+                }
+            }
+
+            Map<String, Object> itemDetails = new HashMap<>();
+            itemDetails.put("productId", item.getProductId());
+            itemDetails.put("quantity", item.getQuantity());
+            itemDetails.put("total", Math.max(itemTotal, 0));
+            items.put("item_" + i, itemDetails);
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider extracting item details mapping logic</b></div>
<div id="fix">

Consider extracting the item details mapping logic into a separate helper method to improve code readability and maintainability. The current implementation mixes business logic with data mapping.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
            items.put("item_" + i, createItemDetails(item, itemTotal));

    private Map<String, Object> createItemDetails(OrderItem item, double itemTotal) {
        Map<String, Object> itemDetails = new HashMap<>();
        itemDetails.put("productId", item.getProductId());
        itemDetails.put("quantity", item.getQuantity());
        itemDetails.put("total", Math.max(itemTotal, 0));
        return itemDetails;
    }
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rkj6124/bito-test/pull/3#issuecomment-2534994613>#cce4ef</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged